cmd/time.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import "time"
+
+var minioTZ = map[string]int{
+
+	"PST": -8 * 3600,
+	"PDT": -7 * 3600,
+
+	"EST": -5 * 3600,
+	"EDT": -4 * 3600,
+
+	"GMT": 0,
+	"UTC": 0,
+
+	"CET":  1 * 3600,
+	"CEST": 2 * 3600,
+}
+
+// parseTime returns the correct time when a timezone is provided
+// with the time string, if the specified timezone is not found in
+// minioTZ, call only the standard golang time parsing.
+func parseTime(layout, value string) (time.Time, error) {
+	// Standard golang time parsing
+	t, err := time.Parse(layout, value)
```

Comment:
My observation on time/timezone in Go is below

* Go has an [issue](https://github.com/golang/go/issues/18012) supporting abbreviated timezone.
* `time.Parse()` sets abbreviated timezone text, not correct value to know time difference with UTC.
* However `time.ParseInLocation()` works properly when timezone is passed as argument.
* As RFC1123 carries abbreviated timezone we have problem with name collision eg. CST refers to multiple timezone.

Based on the above we could do
* No support other than GMT as per RFC.
* If we support other than GMT, we would need to support all of them (Note: there is no solution to abbreviated timezone name collision).

If we go with option 2,  the right fix would just have a map of abbreviated name to full name and use `time.ParseInLocation()` like below
```go
var NameLocationMap = map[string]string{"PST":"US/Pacific", ...}
t, err = time.Parse(time.RFC1123, timeString)
l, err = t.Location()
location, ok = NameLocationMap[l.String()]
if ok {
    l, err = time.LoadLocation(location)
    t, err = time.ParseInLocation(time.RFC1123, timeString, l)
}

fmt.Println("local time:", t.Local())
fmt.Println("UTC time:", t.UTC())
```